src/react-vite/01-intro-to-react/intro-to-react.md

```diff
@@ -8,45 +8,41 @@
 
 ## Overview
 
-TODO
-
-## Objective 1
-
-TODO
-
-### Key concepts
-
-TODO
-
-#### Concept 1
-
-TODO
-
-#### Concept 2
-
-TODO
-
-### Requirements
-
-TODO
-
-### Setup
-
-TODO
-
-### Verify
-
-TODO
-
-### Solution
-
-<details>
-<summary>Click to see the solution</summary>
-
-TODO
-
-</details>
-
-## Next steps
-
-TODO
\ No newline at end of file
+React first appeared in 2013 with "components" as the basic building block for
+constructing a user interface. Initially components were created using
+JavaScript classes which implemented an interface that defined lifecycle
+methods. One of React's notable characteristics has existed from the very
+beginning: the use of [JSX
+syntax](https://react.dev/learn#writing-markup-with-jsx) to define the structure
+of a component.
+
+### Key React Concepts
+
+- Data in a React based application flows one way: into a component through
+`props` or state then the user interface is rendered, user interactions cause
+changes to state, changes to state cause props to be updated and passed to the
+component again - restarting the cycle.
+
+- React's Virtual DOM allows React to quickly determine when the user interface
+needs to be updated; since updates to the browser's DOM are time-consuming only
+making them when necessary is critical to React's rendering performance. When
+the state provided to a React component changes the React framework will invoke
+the component's function or a class component's render method. The output of the
+invocation is compared to the result of the previous invocation - which was
+stored in React's Virtual DOM. If the results are **the same** no action is
+taken, if the results are **different** the Virtual DOM is updated with just the
+changes and then the browser's DOM is similarly updated.
+
+- A major change occurred in 2019 with the introduction of functional components
+and hooks. A component can now be defined using a single function, the component
+can manage its internal data and functionality through the use of
+[hooks](https://react.dev/learn#using-hooks) which are functions that provide
+data to a component or cause the component to render.
+
+- After version 18 was released the React team responded to needs for a
```

Comment:
I would change this to not be a list and just be plain paragraphs. It might help to break them up into little sections with headers, e.g. Unidirectional data flow, Efficiency of the virtual DOM, Rise of functional components and hooks, and Server-side rendering with React Server Components. This would make it easier to scan what they key concepts are.